packages/amplify-graphql-conversation-transformer/src/tools/process-tools.ts

```diff
@@ -234,3 +255,17 @@ const generatePropertyTypes = (
 const isModelListOperation = (toolName: string, responseType: TypeNode): boolean => {
   return getBaseType(responseType).startsWith('Model') && toolName.startsWith('list');
 };
+
+const modelListQueryName = (modelTool: ModelOperationTool, ctx: TransformerContextProvider): string => {
+  const { modelName } = modelTool;
+  const model = getObjectTypeFromName(modelName, ctx);
+
+  const modelDirective = model.directives?.find((directive) => directive.name.value === 'model');
+  const queriesArgument = modelDirective?.arguments?.find((arg) => arg.name.value === 'queries');
+  const queriesValue = queriesArgument?.value as ObjectValueNode | undefined;
+  const listField = queriesValue?.fields?.find((field) => field.name.value === 'list');
+  const listValue = listField?.value as StringValueNode | undefined;
+  const listQueryArgument = listValue?.value;
+
+  return listQueryArgument ?? `list${pluralize(toUpper(modelName))}`;
```

Comment:
For constructing the list query name consistently between this and model transformer, recommend using `getFieldNameFor('list', modelName)` that is exported from the transformer core - https://github.com/aws-amplify/amplify-category-api/blob/acc8140c323988da595630ed9c047e03a825635f/packages/amplify-graphql-transformer-core/src/utils/operation-names.ts#L26